WDIO-Mocha Assignment

madhumitha-bs opened the pull request:
Latest results from Browserstack Automate: https://automate.browserstack.com/builds/63f304fa870dfbc9fc3b502ce167a3e4c5a42af6/sessions/3aa84dff03f9354728d2bfd02e03bd0bf809535c?auth_token=4346f0dbe044962f1c63ba874b56957d8793e38f312fb7e68def8c780320a73d

madhumitha-bs committed:
Adding config for browserstack automate

madhumitha-bs committed:
Removing unwanted changes

madhumitha-bs committed:
Added JUnit Reporters

madhumitha-bs committed:
Commiting to test Trigger in Jenkins Test1

madhumitha-bs committed:
Commiting to test Trigger in Jenkins Test2

madhumitha-bs committed:
Commiting to test Trigger in Jenkins Test2.2

madhumitha-bs committed:
Commiting to test Trigger in Jenkins Test2.3

madhumitha-bs committed:
Commiting to test Trigger in Jenkins Test2.4

madhumitha-bs committed:
Commiting to test Trigger in Jenkins Test2.5

madhumitha-bs committed:
Revert "Removing unwanted changes"

This reverts commit 0a66612e338cfc9e66f7de68698577c9d47c358f.

madhumitha-bs committed:
Removing commented out line

madhumitha-bs committed:
Updating readme.md file

madhumitha-bs committed:
Adding extra lines

madhumitha-bs committed:
Adding extra line to config file

Sameer-Dhole commented on wdio-mocha-practice-assignment/test/pageobjects/home.page.ts:
```diff
@@ -0,0 +1,55 @@
+    public async searchProduct(productName:string){
+        const searchBtn= await this.searchButton;
+        await searchBtn.waitForDisplayed();
+        browser.pause
```
avoid explicit pauses , add awaits if using pauses

Sameer-Dhole commented on wdio-mocha-practice-assignment/wdio.conf.ts:
```diff
@@ -0,0 +1,328 @@
+            'resolution': '1920x1080' 
+        }
+      }],
+    user:  'madhumithavenkat_ZxvnWs',
```
fetch this from env variables and don't commit values

Sameer-Dhole commented on wdio-mocha-practice-assignment/wdio.conf.ts:
```diff
@@ -0,0 +1,328 @@
+            'browserName': 'Chrome',
+            'browserVersion': 'latest',
+            'projectName': 'My WebdriverIO Mocha Test Project',
+            'buildName': 'Build 5 for WebdriverIO Mocha ',
```
make this dynamic

Sameer-Dhole commented on wdio-mocha-practice-assignment/test/specs/test.e2e.ts:
```diff
@@ -0,0 +1,25 @@
+    before(async () => {
+        // Runs before all tests in the suite
+        await HomePage.open()
+        productToSearch = "Laptop"
```
use for each loop to run multiple examples instead od just laptop

madhumitha-bs committed:
Making final changes

Sameer-Dhole commented on wdio-mocha-practice-assignment/test/specs/test.e2e.ts:
```diff
@@ -0,0 +1,25 @@
+        expect(spanText).toContain(" results for \"Laptop\"")
+    })
+
+    after(async () => {
```
no need for this for only console.log

Sameer-Dhole replied:
move this to after block in conf

madhumitha-bs committed:
Making build name dynamic and hiding credentials

madhumitha-bs committed:
Removing browser.pause and unwanted console.logs

madhumitha-bs committed:
Adding for each loop

madhumitha-bs committed:
Removing unwanted console.log

madhumitha-bs commented:
@Sameer-Dhole I have addressed the comments for the PR. Latest build results: https://automate.browserstack.com/dashboard/v2/public-build/NVF4VU5tN21PVkFVSzF5TTB0d2luTzEzMkNQSUNud1dmSnViZWRJYkFJNTZrL3hGN1RyV0JjVUc5c1RTQVBNOEhtRTdVc1k0YlNHZ2Z0ZS9ENjhBV0E9PS0tQzlpWEx2MThhYWFZUGpwcTk0RG1wdz09--f24fdc466161293434aeefb68bea0178198a4c8a

Sameer-Dhole commented on wdio-mocha-practice-assignment/test/pageobjects/results.page.ts:
```diff
@@ -0,0 +1,28 @@
+  public async getSearchResultSpan(): Promise<string> {
+    const spanText = await this.getSpanText;
+    await spanText.waitForDisplayed();
+    return spanText.getText();
```
use awaits on these methods

madhumitha-bs committed:
Adding readme file

madhumitha-bs committed:
Trying webhooks

madhumitha-bs committed:
Adding spanmessage

Sameer-Dhole commented:
LGTM for wdio mocha and jenkins
